src/pages/StyleDetail/StyleDetail.js

```diff
@@ -0,0 +1,239 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+const StyleDetail = () => {
+  const [postPage, setPostPage] = useState([]);
+
+  useEffect(() => {
+    fetch('/data/data.json')
+      .then(response => response.json())
+      .then(result => setPostPage(result));
+  }, []);
+
+  return (
+    <Container>
+      <Content>
+        <SocialPost>
+          {postPage.map(post => {
+            return (
+              <>
+                <UserInfoState key={post.id}>
+                  <UserInfo>
+                    <UserProfileImage src={post.profile_image_url} />
+                    <UserId>{post.nickName}</UserId>
+                  </UserInfo>
+                  <FollowButton>팔로우</FollowButton>
+                </UserInfoState>
+                <PostImage>
+                  <UserPostImage src={post.post_image_url} />
+                </PostImage>
+                <SocialProduct>
+                  <ProductTitle>
+                    <ProductTitleText>
+                      상품 태그 <TextStrong>1</TextStrong>개
+                    </ProductTitleText>
+                  </ProductTitle>
+                </SocialProduct>
+                <ProductList>
+                  <ProductItem>
+                    <ProductImage src={post.products[0].thumbnail_image_url} />
+                  </ProductItem>
+                  <ProductItem>
+                    <ProductName>{post.products[0].en_name}</ProductName>
+                  </ProductItem>
+                  <ProductItem>
+                    <ProductPrice>
+                      {post.products[0].price.toLocaleString()}원
+                    </ProductPrice>
+                  </ProductItem>
+                </ProductList>
+                <SocialContent>
+                  <SocialLikeBox>
+                    <SocialLikeButton>
+                      <FontAwesomeIcon
+                        icon="fa-regular fa-face-smile"
+                        size="2x"
+                        color="gray"
+                      />
+                    </SocialLikeButton>
+                    <SocialLikeButtonText>
+                      공감<TextStrong>{post.likes}</TextStrong>개
+                    </SocialLikeButtonText>
+                  </SocialLikeBox>
+                  <ShareButton>
+                    <FontAwesomeIcon
+                      icon="fa-regular fa-share-from-square"
+                      size="2x"
+                      color="gray"
+                    />
+                  </ShareButton>
+                </SocialContent>
+                <SocialText>{post.feed_text}</SocialText>
+              </>
+            );
+          })}
```

Comment:
이거 컴포넌트로 분리하고 싶은 욕구가 생기지 않으신지요? 👀👀👀

```suggestion
          {postPage.map(post => <StyleCard {...post} />)}
```

Reply:
네! 멘토님 덕분에 많이 배웠습니다. 감사합니다 :)